operator/v1/types_machineconfiguration.go

```diff
@@ -65,8 +65,71 @@ type MachineConfigurationStatus struct {
 	// +openshift:enable:FeatureGate=NodeDisruptionPolicy
 	// +optional
 	NodeDisruptionPolicyStatus NodeDisruptionPolicyStatus `json:"nodeDisruptionPolicyStatus"`
+
+	// nodeSynchronizersStatus is the status of the machines managed by the node synchronizers.
+	// +openshift:enable:FeatureGate=PinnedImages
+	// +listType=map
+	// +listMapKey=name
+	// +optional
+	NodeSynchronizersStatus []NodeSynchronizerStatus `json:"nodeSynchronizersStatus"`
 }
 
+// +kubebuilder:validation:XValidation:rule="self.machineCount >= self.updatedMachineCount", message="machineCount must be greater than or equal to updatedMachineCount"
+// +kubebuilder:validation:XValidation:rule="self.machineCount >= self.availableMachineCount", message="machineCount must be greater than or equal to availableMachineCount"
+// +kubebuilder:validation:XValidation:rule="self.machineCount >= self.unavailableMachineCount", message="machineCount must be greater than or equal to unavailableMachineCount"
+// +kubebuilder:validation:XValidation:rule="self.machineCount >= self.readyMachineCount", message="machineCount must be greater than or equal to readyMachineCount"
+// +kubebuilder:validation:XValidation:rule="self.availableMachineCount >= self.readyMachineCount", message="availableMachineCount must be greater than or equal to readyMachineCount"
+type NodeSynchronizerStatus struct {
+	// name is the name of the node synchronizer.
+	// Must be a lowercase RFC-1123 hostname (https://tools.ietf.org/html/rfc1123)
+	// It may consist of only alphanumeric characters, hyphens (-) and periods (.)
+	// and must be at most 253 characters in length.
+	// +kubebuilder:validation:MaxLength=253
+	// +kubebuilder:validation:Pattern=`^([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9])(\.([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]{0,61}[a-zA-Z0-9]))*$`
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+	// nodeSynchronizerType is the name of the node synchronizer type.
+	// +kubebuilder:validation:Required
+	NodeSynchronizerType NodeSynchronizerType `json:"nodeSynchronizerType"`
+	// nodeSelector specifies a label selector for Machines that match this synchronizer.
+	// +optional
+	NodeSelector *metav1.LabelSelector `json:"nodeSelector,omitempty"`
```

Comment:
since the existing synchronizers are pool based I considered having a pool name in status but the node selector fells my flexible and future proof. So for example if a synchronizer is defined on the pool level then that label should be defined here.

Reply:
```
          nodeSynchronizersStatus:
            - name: master-pool
              nodeSynchronizerType: PinnedImageSets
              nodeSelector:
                matchLabels:
                  machineconfiguration.openshift.io/role: "master"
              machineCount: 3
              updatedMachineCount: 3
              readyMachineCount: 3
              availableMachineCount: 3
              unavailableMachineCount: 0
              observedGeneration: 1
  ```